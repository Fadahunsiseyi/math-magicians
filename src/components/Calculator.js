/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import './calculator.css';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  render() {
    return (
      <div className="calculator-container">
        <div className="answer-div">
          <span>0</span>
        </div>
        <div className="first-div">
          <button type="button" className="btn clear-all number">AC</button>
          <button type="button" className="btn number">+/-</button>
          <button type="button" className="btn modulus number">%</button>
          <button type="button" className="btn divide operator">รท</button>
        </div>
        <div className="second-div">
          <button type="button" className="btn number">7</button>
          <button type="button" className="btn number">8</button>
          <button type="button" className="btn number">9</button>
          <button type="button" className="btn multiplication operator">x</button>
        </div>
        <div className="third-div">
          <button type="button" className="btn number">4</button>
          <button type="button" className="btn number">5</button>
          <button type="button" className="btn number">6</button>
          <button type="button" className="btn substraction operator">-</button>
        </div>
        <div className="fourth-div">
          <button type="button" className="btn number">1</button>
          <button type="button" className="btn number">2</button>
          <button type="button" className="btn number">3</button>
          <button type="button" className="btn addition operator">+</button>
        </div>
        <div className="fourth-div">
          <button type="button" className="btn number zero">0</button>
          <button type="button" className="btn period number">.</button>
          <button type="button" className="btn equals-to operator">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
