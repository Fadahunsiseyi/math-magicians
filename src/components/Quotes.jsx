import React from 'react';

const Quotes = () => (
  <section>
    <p>
      Mathematics is not about numbers,equations,computations or algorithms:
      it is also about understanding
    </p>
    <p>
      <strong>â€” William Paul Thurston</strong>
    </p>
  </section>
);

export default Quotes;
