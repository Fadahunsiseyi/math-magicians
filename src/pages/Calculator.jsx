import React from 'react'

const Calculator = () => {
  return (
    <div>
      welcome to the Calculator
    </div>
  )
}

export default Calculator
