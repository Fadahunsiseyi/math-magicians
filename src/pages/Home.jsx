import React from 'react'

const Home = () => {
  return (
    <div>
      welcome to the home page
    </div>
  )
}

export default Home
