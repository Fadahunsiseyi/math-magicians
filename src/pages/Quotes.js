import React from 'react';

const Quotes = () => (
  <div>
    welcome to the quote page
  </div>
);

export default Quotes;
