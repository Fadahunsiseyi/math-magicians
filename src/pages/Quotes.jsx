import React from 'react'

const Quotes = () => {
  return (
    <div>
      welcome to the quote page
    </div>
  )
}

export default Quotes
