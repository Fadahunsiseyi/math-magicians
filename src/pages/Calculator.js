import React from 'react';
import Calculator from '../components/Calculator';

const CalculatorPage = () => (
  <div>
    <Calculator />
  </div>
);

export default CalculatorPage;
