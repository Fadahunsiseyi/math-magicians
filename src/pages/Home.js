import React from 'react';
import Calculator from '../components/Calculator';

const Home = () => (
  <div>
    <Calculator />
  </div>
);

export default Home;
