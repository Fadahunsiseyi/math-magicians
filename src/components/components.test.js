import renderer from 'react-test-renderer';
import React from 'react';
import Home from '../pages/Home';
import Calculator from '../pages/Calculator';
import Quotes from '../pages/Quotes';

it('renders correctly', () => {
  const testHome = renderer.create(<Home />);
  const testJson = testHome.toJSON();
  expect(testJson).toMatchSnapshot();
});

it('renders correctly', () => {
  const testCalculator = renderer.create(<Calculator />);
  const testJson = testCalculator.toJSON();
  expect(testJson).toMatchSnapshot();
});

it('renders correctly', () => {
  const testQuotes = renderer.create(<Quotes />);
  const testJson = testQuotes.toJSON();
  expect(testJson).toMatchSnapshot();
});
