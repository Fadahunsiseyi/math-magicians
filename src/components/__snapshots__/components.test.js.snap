// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section>
    <h5
      className="title"
    >
      Welcome to our page
    </h5>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Molestie at elementum eu facilisis. Pretium quam vulputate dignissim suspendisse in est ante in nibh. Velit aliquet sagittis id consectetur purus ut faucibus. Lectus proin nibh nisl condimentum id venenatis a. Commodo ullamcorper a lacus vestibulum sed arcu non. Pretium fusce id velit ut tortor pretium. Mauris nunc congue nisi vitae. Sed lectus vestibulum mattis ullamcorper velit. Gravida neque convallis a cras semper.
    </p>
    <p>
      Pellentesque sit amet porttitor eget dolor morbi non arcu risus. Aliquam sem fringilla ut morbi. In tellus integer feugiat scelerisque varius morbi enim nunc. Arcu ac tortor dignissim convallis aenean et. Sagittis orci a scelerisque purus semper. Commodo quis imperdiet massa tincidunt nunc. Egestas purus viverra accumsan in nisl nisi scelerisque eu ultrices. Elementum sagittis vitae et leo duis ut diam quam nulla. Commodo sed egestas egestas fringilla phasellus faucibus scelerisque. Lacus laoreet non curabitur gravida arcu ac tortor dignissim. Dolor sit amet consectetur adipiscing elit ut aliquam purus. Viverra accumsan in nisl nisi. Fusce ut placerat orci nulla. Vitae tortor condimentum lacinia quis vel eros donec ac odio. Venenatis tellus in metus vulputate eu. Purus non enim praesent elementum facilisis leo.
    </p>
  </section>
</div>
`;

exports[`renders correctly 2`] = `
<div>
  <div
    className="calc-container"
  >
    <div
      className="title"
    >
      <h3>
        Let do some maths!
      </h3>
    </div>
    <div
      className="calculator-container"
      onChange={[Function]}
    >
      <div
        className="answer-div"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="first-div"
      >
        <button
          className="btn clear-all number"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn modulus number"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn divide operator"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="second-div"
      >
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn multiplication operator"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="third-div"
      >
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn substraction operator"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="fourth-div"
      >
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn number"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn addition operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="fourth-div"
      >
        <button
          className="btn number zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn period number"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn equals-to operator"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div>
  <section
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <p>
      Mathematics is not about numbers,equations,computations or algorithms: it is also about understanding
    </p>
    <p>
      <strong>
        — William Paul Thurston
      </strong>
    </p>
  </section>
</div>
`;
